client/state/data-layer/wpcom/comments/index.js

```diff
@@ -156,6 +160,35 @@ export const announceFailure = ( { dispatch, getState }, { siteId, postId } ) =>
 	dispatch( errorNotice( error ) );
 };
 
+export const deleteComment = ( { dispatch }, action ) => {
+	const { siteId, commentId } = action;
+	dispatch(
+		http( {
+			method: 'POST',
+			apiVersion: '1.1',
+			path: `/sites/${ siteId }/comments/${ commentId }/delete`,
+			onSuccess: {
+				...action,
+				type: COMMENTS_DELETE_SUCCESS,
+			},
+			onFailure: {
+				...action,
+				type: COMMENTS_DELETE_FAILURE,
+			},
+		} )
+	);
+};
+
+export const removeCommentFromState = ( { dispatch }, action ) => {
+	const { siteId, postId, commentId } = action;
+	dispatch( removeComment( siteId, postId, commentId ) );
```

Comment:
Currently `COMMENTS_DELETE` doesn't pass through the reducer, while `COMMENTS_REMOVE` [does](https://github.com/Automattic/wp-calypso/blob/master/client/state/comments/reducer.js#L72-L76).
I could add `_DELETE` as a fall through of `_REMOVE`?

Also: I've noticed that in both the success and failure functions, the dispatches don't work.
The functions are indeed called (a console log in there works fine) `removeComment` and `errorNotice` are never fired (at least according to the Redux Dev Tools).